src/rdkafka_topic.c

```diff
@@ -310,6 +313,11 @@ shptr_rd_kafka_itopic_t *rd_kafka_topic_new0 (rd_kafka_t *rk,
 	if (rkt->rkt_conf.compression_codec == RD_KAFKA_COMPRESSION_INHERIT)
 		rkt->rkt_conf.compression_codec = rk->rk_conf.compression_codec;
 
+        if (rk->rk_type == RD_KAFKA_PRODUCER)
+                rd_avg_init(&rkt->rkt_avg_batchsize, RD_AVG_GAUGE, 0,
```

Comment:
i've changed my mind on what I think the most useful batch metric is - I think it's number of messages, not total batch size. number of messages directly informs on the degree of batching, which I think is what i'd mostly be interested in. message sizes vary - you can't immediately tell whether / how much batching is occurring without further investigation.